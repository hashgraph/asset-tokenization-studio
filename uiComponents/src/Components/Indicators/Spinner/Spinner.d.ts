/// <reference types="react" />
import type { SpinnerProps as ChakraSpinnerProps } from "@chakra-ui/react";
export interface SpinnerProps extends ChakraSpinnerProps {
}
export declare const Spinner: (props: SpinnerProps) => JSX.Element;
