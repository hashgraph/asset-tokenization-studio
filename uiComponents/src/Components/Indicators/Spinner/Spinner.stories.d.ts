declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./Spinner").SpinnerProps>;
export default meta;
export declare const XXS: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Spinner").SpinnerProps>;
export declare const XS: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Spinner").SpinnerProps>;
export declare const Small: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Spinner").SpinnerProps>;
export declare const Medium: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Spinner").SpinnerProps>;
export declare const Large: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./Spinner").SpinnerProps>;
