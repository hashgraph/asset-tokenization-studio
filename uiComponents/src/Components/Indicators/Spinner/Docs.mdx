import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as SpinnerStories from "./Spinner.stories";

<Meta of={SpinnerStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2053-17926&t=3ozoVrWQYvgw1fj3-4"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

Add the following configurations to the `theme.tsx` and `Spinner.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Spinner: ConfigSpinner,
  },
};
```

```tsx dark
const ConfigSpinner = {
  baseStyle: {},
  sizes: {},
  defaultProps: {},
};
```

## Props

<ArgTypes of={SpinnerStories} />

## Stories

<Stories title="" />
