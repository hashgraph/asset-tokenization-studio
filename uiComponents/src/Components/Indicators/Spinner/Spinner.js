var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Spinner as ChakraSpinner } from "@chakra-ui/react";
export var Spinner = function (props) {
    return React.createElement(ChakraSpinner, __assign({ "data-testid": "spinner" }, props));
};
