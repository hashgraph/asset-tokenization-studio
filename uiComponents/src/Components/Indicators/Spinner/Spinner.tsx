import React from "react";
import type { SpinnerProps as ChakraSpinnerProps } from "@chakra-ui/react";
import { Spinner as ChakraSpinner } from "@chakra-ui/react";

export interface SpinnerProps extends ChakraSpinnerProps {}

export const Spinner = (props: SpinnerProps) => {
  return <ChakraSpinner data-testid="spinner" {...props} />;
};
