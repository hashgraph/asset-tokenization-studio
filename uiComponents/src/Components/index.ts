export * from "./Basic";
export * from "./Charts";
export * from "./DataDisplay";
export * from "./Forms";
export * from "./Foundations";
export * from "./Indicators";
export * from "./Interaction";
export * from "./Navigation";
export * from "./Overlay";
