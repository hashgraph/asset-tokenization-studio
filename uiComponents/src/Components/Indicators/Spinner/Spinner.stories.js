var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Spinner } from "./Spinner";
var meta = {
    title: "Design System/Indicators/Spinner",
    component: Spinner,
    argTypes: {
        size: {
            control: "select",
            options: ["xxs", "xs", "sm", "md", "lg"],
            description: "Size of the spinner"
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2053-17926&t=3ozoVrWQYvgw1fj3-4"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) {
    return React.createElement(Spinner, __assign({}, args));
};
export var XXS = Template.bind({});
XXS.args = {
    size: "xxs"
};
export var XS = Template.bind({});
XS.args = {
    size: "xs"
};
export var Small = Template.bind({});
Small.args = {
    size: "sm"
};
export var Medium = Template.bind({});
Medium.args = {
    size: "md"
};
export var Large = Template.bind({});
Large.args = {
    size: "lg"
};
