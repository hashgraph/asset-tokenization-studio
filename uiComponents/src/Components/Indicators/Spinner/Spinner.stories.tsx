import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Spinner } from "./Spinner";

const meta = {
  title: "Design System/Indicators/Spinner",
  component: Spinner,
  argTypes: {
    size: {
      control: "select",
      options: ["xxs", "xs", "sm", "md", "lg"],
      description: "Size of the spinner",
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=2053-17926&t=3ozoVrWQYvgw1fj3-4",
    },
    docs: {},
  },
} as Meta<typeof Spinner>;
export default meta;

const Template: StoryFn<typeof Spinner> = (args) => {
  return <Spinner {...args} />;
};

export const XXS = Template.bind({});
XXS.args = {
  size: "xxs",
};

export const XS = Template.bind({});
XS.args = {
  size: "xs",
};

export const Small = Template.bind({});
Small.args = {
  size: "sm",
};

export const Medium = Template.bind({});
Medium.args = {
  size: "md",
};

export const Large = Template.bind({});
Large.args = {
  size: "lg",
};
