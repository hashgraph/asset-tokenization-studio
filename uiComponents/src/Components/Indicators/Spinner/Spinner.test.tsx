import { render } from "@/test-utils";
import { ConfigSpinner } from "@/Theme/Components/Spinner";
import React from "react";
import type { SpinnerProps } from "./Spinner";
import { Spinner } from "./Spinner";

describe(`<Spinner />`, () => {
  const factoryComponent = (props: SpinnerProps = {}) =>
    render(<Spinner {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    const spinnerStyle = getComputedStyle(component.getByTestId("spinner"));
    expect(spinnerStyle.borderWidth).toBe(ConfigSpinner.sizes.xxs.borderWidth);
    expect(component.asFragment()).toMatchSnapshot("Default Spinner");
  });

  test("renders correctly with size", () => {
    const component = factoryComponent({ size: "sm" });
    const spinnerStyle = getComputedStyle(component.getByTestId("spinner"));
    expect(spinnerStyle.borderWidth).toBe(ConfigSpinner.sizes.sm.borderWidth);
    expect(component.asFragment()).toMatchSnapshot("Sized Spinner");
  });
});
