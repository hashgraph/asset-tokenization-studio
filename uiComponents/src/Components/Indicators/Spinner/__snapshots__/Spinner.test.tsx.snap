// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly with size: Sized Spinner 1`] = `
<DocumentFragment>
  <div
    class="chakra-spinner css-1n0uu9w"
    data-testid="spinner"
  >
    <span
      class="css-8b45rq"
    >
      Loading...
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Spinner /> renders correctly: Default Spinner 1`] = `
<DocumentFragment>
  <div
    class="chakra-spinner css-6dnz8x"
    data-testid="spinner"
  >
    <span
      class="css-8b45rq"
    >
      Loading...
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
